Treat '=' character as legal value in SCR_Config

mcfadden8 opened the pull request:
This is to address Issue #432

**Note**: I still need to write a test for this.  But I wanted to add this draft pull request so that the changes might be reviewed/debated.

Previously, the parser in SCR_Config was treating '=' as an invalid
character within a value for a given token.  This caused calling
applications to fail when they passed in directory path names that
contained an '=' character.

This failure behavior is inconsistent because when the application uses
environment variables instead of SCR_Config, this particular character
is allowed.

The fix was to remove the check when the parser is in 'in_value' state.

adammoody commented:
Thanks, @mcfadden8 .  This looks good to me.

It looks like there might be a separate case (before_value) if the value string happens to *start* with an "=" character.
```
char val[] = "=startswith"
SCR_Config("SCR_PREFIX=%s", val)
```
Do you see any issues if we also pick that up?

mcfadden8 committed:
Treat '=' character as legal value in SCR_Config

Previously, the parser in SCR_Config was treating '=' as an invalid
character within a value for a given token.  This caused calling
applications to fail when they passed in directory path names that
contained an '=' character.

This failure behavior is inconsistent because when the application uses
environment variables instead of SCR_Config, this particular character
is allowed.

The fix was to remove the check when the parser is either in 'before_value' or
'in_value' state.

Add tests for SCR_Config with '=' character

Update SCR_Config docs describing '=' char as being valid

mcfadden8 commented:
@adammoody - This PR is ready for your review and merge (when approved).  I covered the additional case you found (thanks!), updated the docs, and updated/ran the tests.

adammoody commented:
Nice work, @mcfadden8 !  Looks good to me.
